src/NCalc.Core/Visitors/FunctionExtractionVisitor.cs

```diff
@@ -0,0 +1,63 @@
+using NCalc.Domain;
+
+namespace NCalc.Visitors;
+
+/// <summary>
+/// Visitor dedicated to extract <see cref="Function"/> names from a <see cref="LogicalExpression"/>.
+/// </summary>
+public sealed class FunctionExtractionVisitor : ILogicalExpressionVisitor<List<string>>
+{
+    public List<string> Visit(Identifier identifier) => [];
+
+    public List<string> Visit(LogicalExpressionList list)
+    {
+        var functions = new List<string>();
+        foreach (var value in list)
+        {
+            if (value is not Function function) 
+                continue;
```

Comment:
Do we really need `is not Function` with `continue` here? Personally, I don't like `continue` as it makes code hard to read. What do you think?

Reply:
[Inverting the `if` increase the nesting:](https://www.jetbrains.com/help/rider/InvertIf.html)
![image](https://github.com/user-attachments/assets/c605fc83-6a10-449c-8384-155a7a379b7f)

But in this case as the code is small, does not make much difference.